### PR TITLE
Add OID type.

### PR DESCRIPTION
**Pull request by stbof**
DOC-3383

### CODE REVIEW

File: v22.1/oid.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+title: OID
+summary: The OID data type stores an unsigned 32 bit integer value.
+toc: true
+docs_area: reference.sql
+---
+
+The `OID` (Object Identifier) [data type](data-types.html) stores an unsigned 32 bit integer value. OIDs are used internally as primary keys for tables in [system schema](system-catalogs.html). OIDs are not used in user-created tables.
```
Comment (ajwerner) [inline v22.1/oid.md:8]: sadly I think `OIDs` can be used in user-created tables.
```
create table baz (k oid primary key);
CREATE TABLE
```

Our OIDs are very similar to postgres's. There are "subtypes" of oids which correspond to different types. It's rather frustratingly complex. https://www.postgresql.org/docs/14/datatype-oid.html

One note is that pg_catalog is synthesized in cockroach whereas it's a real table in postgres. The structure of our OIDs is bespoke for each of the subtypes. We can omit that for now. I'm fine keeping this doc vague, but it is an area which could use further expansion.

To clarify what I mean, a table has a `regclass`. One can cast a string name to a regclass: 

```
CREATE TABLE foo (i INT);
SELECT 'foo'::regclass;
SELECT 'foo'::regclass::oid;
```

```
CREATE TABLE
  regclass
------------
  foo
(1 row)
  oid
-------
  116
(1 row)
```

**Review by ajwerner — COMMENTED** [review]
I'm afraid that my comment is only going to muddy the waters.

File: v22.1/oid.md (hunk 1 above)
Comment (stbof) [reply v22.1/oid.md:8]: @ajwerner How do you want to move this forward?  Shall I just remove the sentence `OIDs are not used in user-created tables.`? What use cases should I document?

Comment (postamar) [reply v22.1/oid.md:8]: I suggest something along the lines of `OIDs should not be used in user-created tables. Values of this type are not guaranteed to be stable across major releases`.

Comment (rafiss) [reply v22.1/oid.md:8]: i think we should also mention that users should not rely on OIDs to be globally unique. each OID subtype is only unique within certain namespace, and it probably isn't worth getting into all the details of that.

**Comment by stbof** [comment]
@ajwerner  Please review use cases and limitations.

**Review by ajwerner — APPROVED** [review]
:lgtm:

I like it! Provides useful information and avoids going into the weeds. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14662#-:-N8VP9oqC4sashwN8JJs:bn2xozx)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajwerner, @postamar, and @rafiss)

File: v22.1/data-types.md [hunk 2]
```diff
@@ -24,6 +24,7 @@ Type | Description | Example
 [`INT`](int.html) | A signed integer, up to 64 bits. | `12345`
 [`INTERVAL`](interval.html) | A span of time.  | `INTERVAL '2h30m30s'`
 [`JSONB`](jsonb.html) | JSON (JavaScript Object Notation) data.  | `'{"first_name": "Lola", "last_name": "Dog", "location": "NYC", "online" : true, "friends" : 547}'`
+[`OID`](oid.html) | An unsigned 32 bit integer. | `4294967295`
```
Comment (taroface) [inline v22.1/data-types.md:27]:
```suggestion
[`OID`](oid.html) | An unsigned 32-bit integer. | `4294967295`
```

File: v22.1/oid.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+---
+title: OID
+summary: The OID data type stores an unsigned 32 bit integer value.
```
Comment (taroface) [inline v22.1/oid.md:3]:
```suggestion
summary: The OID data type stores an unsigned 32-bit integer value.
```

File: v22.1/oid.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+toc: true
+docs_area: reference.sql
+---
+
+The `OID` (Object Identifier) [data type](data-types.html) stores an unsigned 32 bit integer value.
```
Comment (taroface) [inline v22.1/oid.md:8]:
```suggestion
The `OID` (Object Identifier) [data type](data-types.html) stores an unsigned 32-bit integer value.
```

Comment (taroface) [reply v22.1/oid.md:8]: Does "Object Identifier" need to be capitalized? Postgres docs imply no: https://www.postgresql.org/docs/current/datatype-oid.html

File: v22.1/oid.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+OIDs are used internally as primary keys for tables in [system catalogs](system-catalogs.html), notably `information_schema` and `pg_catalog`.
+
+CockroachDB supports many [functions](functions-and-operators.html#built-in-functions) that accept OIDs as argument types and return OIDs, and [operators](functions-and-operators.html#operators) that operate on OIDs and other data types. These functions are used by are used by drivers and ORMs and you can use these functions to introspect schema.
```
Comment (taroface) [inline v22.1/oid.md:12]:
```suggestion
CockroachDB supports many [functions](functions-and-operators.html#built-in-functions) that accept OIDs as argument types and return OIDs, and [operators](functions-and-operators.html#operators) that operate on OIDs and other data types. These functions are used by drivers and ORMs and you can use these functions to introspect your schema.
```

**Review by taroface — APPROVED** [review]
Looks good; a few nits.

File: v22.1/oid.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+---
+title: OID
+summary: The OID data type stores an unsigned 32-bit integer value.
+toc: true
+docs_area: reference.sql
+---
+
+The object identifier (OID) [data type](data-types.html) stores an unsigned 32-bit integer value.
```
Comment (stbof) [inline v22.1/oid.md:8]:
```suggestion
The object identifier (`OID`) [data type](data-types.html) stores an unsigned 32-bit integer value.
```

**Pull request merged**